lib/src/markdown_processor.dart

```diff
@@ -181,12 +184,73 @@ final List<md.BlockSyntax> _markdownBlockSyntaxes = [
 final RegExp _hideSchemes = RegExp('^(http|https)://');
 
 class MatchingLinkResult {
-  final ModelElement element;
+  final ModelElement modelElement;
   final bool warn;
 
-  MatchingLinkResult(this.element, {this.warn = true});
+  MatchingLinkResult(this.modelElement, {this.warn = true});
+
+  @override
+  bool operator ==(Object other) {
+    return other is MatchingLinkResult &&
+        modelElement == other.modelElement &&
+        warn == other.warn;
+  }
+
+  @override
+  int get hashCode => hash2(modelElement, warn);
+
+  bool isEquivalentTo(MatchingLinkResult other) {
+    if (this == other) return true;
+    if (modelElement?.canonicalModelElement ==
+        other.modelElement?.canonicalModelElement) return true;
+    if (other.modelElement is Parameter && modelElement == null) {
+      return true;
+    }
+    if (modelElement is Parameter && other.modelElement == null) {
+      return true;
+    }
+    return false;
+  }
+
+  @override
+  String toString() {
+    return 'element: ${modelElement?.fullyQualifiedName} warn: $warn';
+  }
 }
 
+class _MarkdownStats {
+  int totalReferences = 0;
```

Comment:
Is there any reason to not store these as doubles? Just to simplify the reporting logic.

Reply:
+1; making it a double would be more of a hack than a proper impl.